src/puj/push/unit.clj

```diff
@@ -45,34 +45,41 @@
   (open-count [this] "The number of CodeBlocks to open following the instruction. Used by linear genomes.")
   (required-stacks [this] "A set of stack names relevant to the instruction."))
 
+(defn make-collection
+  "If thing is a collection, returns it. Otherwise, returns it in a vector"
+  [thing]
+  (if (coll? thing)
+    thing
+    [thing]))
 
-; @TODO: Allow for non-collection instruction results.
-
+; 
 (defrecord SimpleInstruction [input-stacks output-stacks opens func]
   PushInstruction
   (open-count [_] opens)
-  (required-stacks [_] (set (concat input-stacks output-stacks)))
+  (required-stacks [_] (set (concat input-stacks
+                                    (if (coll? output-stacks) ; If output-stacks is not a collection, make it one.
```

Comment:
Seems like another good place to use `make-collection`.

Reply:
Not sure how I missed that one. Done!